macros/src/wasm_export/tools.rs

```diff
@@ -98,28 +122,76 @@ mod tests {
     use syn::{parse::Parser, parse_quote};
 
     #[test]
-    fn test_create_function_call() {
-        // static and async
+    fn test_create_function_call_unified_static_method_async() {
         let stream = TokenStream::from_str(r#"(arg1, arg2): (String, u8)"#).unwrap();
         let inputs = Punctuated::<FnArg, Comma>::parse_terminated
             .parse2(stream)
             .unwrap();
         let fn_name = Ident::new("some_name", Span::call_site());
         let is_async = true;
-        let result = create_function_call(&fn_name, &inputs, is_async);
+        let result =
+            create_function_call_unified(&fn_name, &inputs, is_async, FunctionContext::Method);
+        // Expected: Self::some_name(arg1, arg2).await.into() -> Note: parse_quote! needs parentheses around tuple args
         let expected: Block = parse_quote!({ Self::some_name((arg1, arg2)).await.into() });
-        assert_eq!(result, expected);
+        assert_eq!(format!("{:?}", result), format!("{:?}", expected)); // Compare string representation for complex Block types
```

Comment:
Absolutely not critical but clippy would suggest 

```suggestion
        assert_eq!(format!("{result:?}"), format!("{expected:?}")); // Compare string representation for complex Block types
```

Reply:
hm we actually run clippy on CI